### PR TITLE
Ensemble Non-default Config Generation

### PR DESCRIPTION
**Pull request by nv-braf**
Generate RunConfigs for the non-default configs of ensemble models.

I've added unit testing and I have run an ensemble model through the profile step and it did complete successfully:

image

### CODE REVIEW

File: model_analyzer/config/generate/quick_run_config_generator.py [hunk 1]
```diff
@@ -280,26 +282,105 @@ def _get_coordinate_values(self, coordinate: Coordinate,
...
+        for model in self._models:
+            mrc, model_index = self._get_next_model_run_config(
+                model, model_index)
             run_config.add_model_run_config(mrc)
 
         return run_config
 
-    def _get_next_model_run_config(self, model_num: int) -> ModelRunConfig:
-        mc = self._get_next_model_config(model_num)
+    def _get_next_model_run_config(
```
Comment (tgerdesnv) [inline model_analyzer/config/generate/quick_run_config_generator.py:293]: this function does a few different things. Can you refactor it?

File: model_analyzer/config/generate/quick_run_config_generator.py [hunk 2]
```diff
@@ -280,26 +282,105 @@ def _get_coordinate_values(self, coordinate: Coordinate,
...
-        model_variant_name = mc.get_field('name')
-        pac = self._get_next_perf_analyzer_config(model_variant_name, model_num)
+        return (ensemble_subconfigs, model_index)
 
-        model_name = self._models[model_num].model_name()
-        return ModelRunConfig(model_name, mc, pac)
+    def _get_next_ensemble_param_combo(self, start_model_index: int,
+                                       end_model_index: int) -> dict:
+        """
+        For the ensemble model the only parameter we need to set
```
Comment (tgerdesnv) [inline model_analyzer/config/generate/quick_run_config_generator.py:341]: Have you confirmed this statement with Iman?

File: model_analyzer/config/generate/quick_run_config_generator.py [hunk 3]
```diff
@@ -280,26 +282,105 @@ def _get_coordinate_values(self, coordinate: Coordinate,
...
-    def _get_next_model_config(self, model_num: int) -> ModelConfig:
+    def _get_next_model_config(self, model: ModelProfileSpec,
+                               dimension_index: int) -> ModelConfig:
         dimension_values = self._get_coordinate_values(
-            self._coordinate_to_measure, model_num)
+            self._coordinate_to_measure, dimension_index)
+
+        if model.cpu_only():
+            kind = "KIND_CPU"
+        elif 'instance_group' in model.get_default_config():
```
Comment (tgerdesnv) [inline model_analyzer/config/generate/quick_run_config_generator.py:378]: Is this the behavior we want? I think the old code would put the model on the GPU unless the config told us we can't.

File: model_analyzer/config/generate/quick_run_config_generator.py (hunk 2 above)
Comment (nv-braf) [reply model_analyzer/config/generate/quick_run_config_generator.py:341]: Yes

File: model_analyzer/config/generate/quick_run_config_generator.py (hunk 1 above)
Comment (nv-braf) [reply model_analyzer/config/generate/quick_run_config_generator.py:293]: It does two things: creates the next MRC and returns the next model count index. I thought about breaking out the model count index, but it's needed when creating the MRC. I'm not sure how I could refactor this method further.

File: model_analyzer/config/generate/quick_run_config_generator.py [hunk 4]
```diff
@@ -280,26 +282,105 @@ def _get_coordinate_values(self, coordinate: Coordinate,
...
             run_config.add_model_run_config(mrc)
 
         return run_config
 
-    def _get_next_model_run_config(self, model_num: int) -> ModelRunConfig:
-        mc = self._get_next_model_config(model_num)
+    def _get_next_model_run_config(
+            self, model: ModelProfileSpec,
+            model_index: int) -> Tuple[ModelRunConfig, int]:
+        start_model_index = deepcopy(model_index)
```
[the remark this answers is not in the document]
Comment (nv-braf) [reply model_analyzer/config/generate/quick_run_config_generator.py:296]: Removed

File: model_analyzer/config/generate/quick_run_config_generator.py (hunk 3 above)
Comment (nv-braf) [reply model_analyzer/config/generate/quick_run_config_generator.py:378]: Talked w/ Iman and I'm reverting this change

**Review by tgerdesnv — APPROVED** [review]

**Pull request merged**